src/include/interval_set.h

```diff
@@ -461,23 +458,30 @@ class interval_set {
     assert(&b != this);
     clear();
 
-    typename std::map<T,T>::const_iterator pa = a.m.begin();
-    typename std::map<T,T>::const_iterator pb = b.m.begin();
-    
-    while (pa != a.m.end() && pb != b.m.end()) {
-      // passing?
-      if (pa->first + pa->second <= pb->first) 
-        { pa++;  continue; }
-      if (pb->first + pb->second <= pa->first) 
-        { pb++;  continue; }
-      T start = MAX(pa->first, pb->first);
-      T en = MIN(pa->first+pa->second, pb->first+pb->second);
-      assert(en > start);
-      insert(start, en-start);
-      if (pa->first+pa->second > pb->first+pb->second)
-        pb++;
-      else
-        pa++; 
+    const interval_set *s;
+    const interval_set *l;
+    typename std::map<T,T>::const_iterator ps, pl;
+
+    if (a.size() < b.size()) {
+      ps = a.m.begin();
+      s = &a;
+      l = &b;
+    }
```

Comment:
Minor style nit, else should be on the same line as the closing brace of the associated if.